src/device/device.go

```diff
@@ -47,6 +49,18 @@ func FromNewDevice(newDeviceArgs model.NewDevice) model.Device {
 	return newDevice
 }
 
+// updateDeviceFromDevice builds a model.UpdateDevice from a model.Device.
+func UpdateDeviceFromDevice(d *model.Device) model.UpdateDevice {
+	return model.UpdateDevice{
```

Comment:
I think there are some struct merge packages that make this a bit more maintainable